doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -436,34 +447,78 @@ is_midplane_reachable(self):
 #### 3.1.5 ModuleBase class new APIs
 
 ##### 3.1.5.1 Need for consistent storage and access of DPU reboot cause, state and health
-1.  The smartswitch needs to know the reboot cause for DPUs. Please refer to the CLI section for the various options and their effects when executed on the switch and DPUs. 
-
-* Each DPU will update its reboot cause history in the Switch ChasissStateDB upon boot up. The recent reboot-cause can be derived from that list of reboot-causes.
-* The get_reboot_cause will return the current reboot-cause of the module.
-* For persistent storage of the DPU reboot-cause and reboot-caue-history files use the existing host storage path and mechanism.
-
-#### Schema for REBOOT_CAUSE - switch stateDB
+#### Reboot Cause
+1. The smartswitch needs to know the reboot cause for DPUs. Please refer to the CLI section for the various "show reboot-cause" options and their effects.
+    * Each DPU will update its reboot cause history in the Switch ChassisStateDB upon boot up and also persist this on the host. The recent reboot-cause is derived from that list of reboot-causes.
+    * The get_reboot_cause will return the current reboot-cause of the module.
+    * For persistent storage of the DPU reboot-cause and reboot-cause-history files use the existing host storage path and mechanism under "/host/reboot-cause/module/dpux".
+    * The storage and retrieval of the reboot-cause of the Switch and PDUs are shown in the sequence diagram
+
+<p align="center"><img src="./images/dpu-reboot-seq.svg"></p>
+
+* The switch boots up. Determines the NPU reboot cause. 
+* Processes the previously stored NPU and DPU reboot-cause files and history files and updates the NPU reboot-cause into the StateDB and the DPU reboot-cause into the ChassisStateDB.
+* The above process is a one-shot event on boot up.
+* The module_db_update function in the NPU-PMON chassisd is an existing function constantly updating the operational status of the DPUs.  This function looks for DPU operational status change events and when the DPUs come out of "offline" state, issues "get_reboot_cause" API to the platform.
```

Comment:
@rameshraghupathy can you state explicityl that out of offline state must mean DPU rebooted just to make sure DPU offline -> online does not happen without reboot.

Reply:
@prgeor  Done

Reply:
@prgeor Done